Sources/Application/TokenHolder.swift

```diff
@@ -1,9 +1,14 @@
 import Authentication
+import Networking
 
-class TokenHolder {
+class TokenHolder: AuthenticationProvider {
     var tokenResponse: TokenResponse?
     var accessToken: String?
     
+    var bearerToken: String {
+        accessToken ?? ""
+    }
```

Comment:
@jdubey @mcgarj Might be worth having a chat about this tomorrow.
As we need to initialise the NetworkClient on successful first time user _and_ successful returning user but we only get a `TokenResponse` (conforming to `AuthenticationProvider`) at the end of the first time user flow, this is a workaround.
I'm sure there's a more considered way to do this looking into the future but this seemed to be the only workable halfway house in my mind.